pkg/app/elasticsearch.go

```diff
@@ -0,0 +1,228 @@
+		Successful int `json:"successful"`
+		Skipped    int `json:"skipped"`
+		Failed     int `json:"failed"`
+	} `json:"_shards"`
+	Hits struct {
+		Total struct {
+			Value    int    `json:"value"`
+			Relation string `json:"relation"`
+		} `json:"total"`
+		MaxScore interface{}   `json:"max_score"`
+		Hits     []interface{} `json:"hits"`
+	} `json:"hits"`
+}
+
+type ElasticsearchInstance struct {
+	cli              kubernetes.Interface
+	namespace        string
+	name             string
+	indexname        string
+	chart            helm.ChartInfo
+	elasticsearchURL string
+}
+
+func NewElasticsearchInstance(name string) App {
+	return &ElasticsearchInstance{
+		name:      name,
+		namespace: "es-test",
+		indexname: "testindex",
+		chart: helm.ChartInfo{
+			Release:  name,
+			RepoURL:  helm.ElasticRepoURL,
+			Chart:    "elasticsearch",
+			RepoName: helm.ElasticRepoName,
+			Version:  "7.4.1",
+			Values: map[string]string{
+				"antiAffinity": "sort",
+				"replicas":     "1",
+			},
+		},
+		elasticsearchURL: "localhost:9200",
+	}
+}
+
+func (esi *ElasticsearchInstance) Init(ctx context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	esi.cli, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (esi *ElasticsearchInstance) Install(ctx context.Context, namespace string) error {
+	esi.namespace = namespace
+	// Get the HELM cli
+	cli := helm.NewCliClient(helm.V3)
+
+	log.Print("Installing the application using helm.", field.M{"app": esi.name})
+	err := cli.AddRepo(ctx, esi.chart.RepoName, esi.chart.RepoURL)
+	if err != nil {
+		return err
+	}
+
+	err = cli.Install(ctx, fmt.Sprintf("%s/%s", esi.chart.RepoName, esi.chart.Chart), esi.chart.Version, esi.name, esi.namespace, esi.chart.Values)
+	if err != nil {
+		return err
+	}
+	log.Print("Application was installed succcessfully.", field.M{"app": esi.name})
+	return nil
+}
+
+func (esi *ElasticsearchInstance) IsReady(ctx context.Context) (bool, error) {
+	log.Print("Waiting for the application to be ready.", field.M{"app": esi.name})
+	ctx, cancel := context.WithTimeout(ctx, esWaitTimeout)
+	defer cancel()
+
+	err := kube.WaitOnStatefulSetReady(ctx, esi.cli, esi.namespace, fmt.Sprintf("%s-master", esi.name))
+	if err != nil {
+		return false, err
+	}
+
+	log.Print("Application is ready.", field.M{"app": esi.name})
+	return true, nil
+}
+
+func (esi *ElasticsearchInstance) Object() crv1alpha1.ObjectReference {
+	return crv1alpha1.ObjectReference{
+		Kind:      "StatefulSet",
+		Name:      fmt.Sprintf("%s-master", esi.name),
+		Namespace: esi.namespace,
+	}
+}
+
+func (esi *ElasticsearchInstance) Uninstall(ctx context.Context) error {
+	cli := helm.NewCliClient(helm.V3)
+	log.Print("UnInstalling the application using helm.", field.M{"app": esi.name})
+
+	err := cli.Uninstall(ctx, esi.name, esi.namespace)
+	if err != nil {
+		errors.Wrapf(err, "Error uninstalling the application %s", esi.name)
```

Comment:
@viveksinghggits you are not storing the value returned by errors.Wrapf(). We can just to `return errors.Wrapf()` here